app/src/organisms/ODD/ProtocolSetup/ProtocolSetupModulesAndDeck/ModuleTableItem.tsx

```diff
@@ -0,0 +1,358 @@
+import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import {
+  ALIGN_CENTER,
+  BORDERS,
+  Chip,
+  COLORS,
+  DeckInfoLabel,
+  DIRECTION_COLUMN,
+  Flex,
+  JUSTIFY_SPACE_BETWEEN,
+  LegacyStyledText,
+  SPACING,
+  TYPOGRAPHY,
+} from '@opentrons/components'
+import {
+  ABSORBANCE_READER_TYPE,
+  FLEX_STACKER_MODULE_TYPE,
+  getModuleDisplayName,
+  getModuleType,
+  TC_MODULE_LOCATION_OT3,
+  THERMOCYCLER_MODULE_TYPE,
+} from '@opentrons/shared-data'
+
+import { SmallButton } from '/app/atoms/buttons'
+import {
+  getFlexStackerPrepCommands,
+  getModulePrepCommands,
+} from '/app/local-resources/modules'
+import { OddInfoScreen } from '/app/molecules/ODDInfoScreen'
+import { OddModal } from '/app/molecules/OddModal'
+import { LocationConflictModal } from '/app/organisms/LocationConflictModal'
+import { ModuleWizardFlows } from '/app/organisms/ModuleWizardFlows'
+import { useToaster } from '/app/organisms/ToasterOven'
+import { getModuleTooHot } from '/app/transformations/modules'
+
+import type { Dispatch, SetStateAction } from 'react'
+import type { AttachedModule, CommandData } from '@opentrons/api-client'
+import type { CutoutConfig, DeckDefinition } from '@opentrons/shared-data'
+import type { ModulePrepCommandsType } from '/app/local-resources/modules'
+import type { ProtocolCalibrationStatus } from '/app/resources/runs'
+import type { AttachedProtocolModuleMatch } from '/app/transformations/analysis'
+
+export type ModuleStatusType =
+  | 'locationConflict'
+  | 'disconnected'
+  | 'shuttleMissing'
+  | 'calibrationBlocked'
+  | 'needsCalibration'
+  | 'needsHome'
+  | 'connected' // when the module is connected and ready
+
+export const getModuleDisplayStatus = (
+  attachedModule: AttachedModule | null,
+  conflictedFixture: CutoutConfig | null,
+  calibrationStatus: ProtocolCalibrationStatus
+): ModuleStatusType => {
+  // deck location conflict
+  if (conflictedFixture !== null) {
```

Comment:
```suggestion
  if (conflictedFixture != null) {
```
unless we don't want to hit this if statement if `conflictedFixture` is undefined!